laravel/resources/views/modals/instance/modal-edit.blade.php

```diff
@@ -13,74 +13,89 @@
+                                       aria-describedby="validationServerQueryPasswordHelp validationServerQueryPasswordFeedback"
+                                       name="serverquery_password" required>
+                                <div id="validationServerQueryPasswordHelp" class="form-text">The password of the previous defined ServerQuery username</div>
                                 <div class="valid-feedback">{{ __("Looks good!") }}</div>
-                                <div class="invalid-feedback">{{ __("Please provide a valid password.") }}</div>
+                                <div id="validationServerQueryPasswordFeedback" class="invalid-feedback">{{ __("Please provide a valid password.") }}</div>
                             </div>
                             <div class="mb-3">
                                 <label class="form-label fw-bold" for="validationClientNickname">Client Nickname</label>
-                                <input type="text" class="form-control" id="validationClientNickname" maxlength="30" name="client_nickname" value="{{ old('client_nickname', $instanceModal->client_nickname) }}" required>
-                                <div class="form-text">How this client should be named on your TeamSpeak server. (Maximum length: 30 characters)</div>
+                                <input type="text" class="form-control" id="validationClientNickname" maxlength="30" name="client_nickname"
+                                       aria-describedby="validationClientNicknameHelp validationClientNicknameFeedback"
+                                       value="{{ old('client_nickname', $instanceModal->client_nickname) }}" required>
+                                <div id="validationClientNicknameHelp" class="form-text">How this client should be named on your TeamSpeak server. (Maximum length: 30 characters)</div>
                                 <div class="valid-feedback">{{ __("Looks good!") }}</div>
-                                <div class="invalid-feedback">{{ __("Please provide a valid nickname.") }}</div>
+                                <div id="validationClientNicknameFeedback" class="invalid-feedback">{{ __("Please provide a valid nickname.") }}</div>
                             </div>
                             <div class="mb-3">
                                 <label for="validationDefaultChannelId" class="form-label fw-bold">Default Channel</label>
-                                <select class="form-select" name="default_channel_id" id="validationDefaultChannelId" aria-describedby="defaultChannelIdHelp">
+                                @if(isset($channel_list[$instanceModal->id]['error']))
+                                    <div class="alert alert-danger" role="alert">
+                                        {{ $channel_list[$instanceModal->id]['error'] }}
+                                    </div>
+                                @else
```

Comment:
[CONCERN]

I'm not a fan of fully removing this input field when the application was unable to retrieve a current channel list.

First of all: This has an effect on the request validation - the field is sometimes defined and sometimes not.

Secondly, you also need a proper code to handle the patch / update in the database. When do you set the default channel to `null` (undefined) and when do you leave it as it was?

Example:

1. User successfully adds a new TeamSpeak instance
2. User successfully sets the default channel to ID 1337 (e. g. "Bots" channel)
3. User edits the ServerQuery credentials as he changed them a few seconds ago on the TeamSpeak server

The channel list can't be retrieved now as the credentials are invalid. However, this is an expected behaviour for the user, so he only wants to update the ServerQuery credentials without changing anything else.

In your case here, the old configured channel ID won't be displayed at all and instead an error message is shown, that the channel list couldn't be retrieved.

In best case, the user can simply update the credentials without changing any other instance settings and afterwards, the channel list can be successfully retrieved again.

Reply:
According to my understandig is when a default channel is set != null and you cant receive the Channel List then you cant show the current default channel name because you havn't the needed channel list from the TS Server. 

```
@foreach($channel_list[$instanceModal->id]['channel_list'] as $channel)
  @if (old('default_channel_id', $instanceModal->default_channel_id) == $channel->cid) "selected"
    <option value="{{ $channel->cid }}" selected>{{ $channel->channel_name }}</option>
  @else
    <option value="{{ $channel->cid }}">{{ $channel->channel_name }}</option>
  @endif
@endforeach
```
This loop can describes that.
I confirmed your mention to not delete the `<select>` input field but what should show in the default option tag? The default channel id where stored in the Model as value and name?

A other solution can be to handle it in the Controller self. When the Key exists then use the value from the key else check if model has an value != null else set null.
https://laravel.com/docs/10.x/validation#validating-when-present